aflr/api_resources/file.py

```diff
@@ -1,19 +1,60 @@
+from posixpath import basename
 import aflr
-from aflr.api_request import APIRequest
+from aflr.helper_classes import ListableResource
+from tkinter import Tk
+from tkinter.filedialog import askopenfilename
+import os
+import requests
+import json
 
 
-class File(APIRequest):
+class File(ListableResource):
     OBJECT_NAME = "file"
+    resource_path = "/file"
+    custom_audio_upload = "/file/customaudio/uploadurl?filename="
+    by_tag = "/file/customaudio/tags?tags="
+    by_metaId = "/file/customaudio/mediaid?mediaId="
+    by_org = "/file/customaudio/org"
 
-    def __init__(self):
-        super().__init__()
-        self.url = self.api_base + "/file"
+    @classmethod
+    def upload_audiofile(cls):
+        # get presigned URL
+        payload = askopenfilename()
+        filename = os.path.basename(payload)
+        headers = {"Content-Type": "audio/mpeg"}
+        url = cls._get_request(path_param=cls.custom_audio_upload + filename)
 
-    def config_test(self):
-        return f"Configured to transact {self.OBJECT_NAME} objects to {self.url} with api_key = {self.api_key}"
+        mediaId = url["mediaId"]
+        fileUploadUrl = url["fileUploadUrl"]
 
-    # get speech files
-    # TBD
+        # Do put request with presigned URL
+        response = requests.request(
+            "PUT", url=fileUploadUrl, headers=headers, data=payload
+        )
 
-    # get sound files
-    # TBD
+        response = {
+            "message": f" Success. Please make sure to save this mediaId : {mediaId}"
+        }
+        return response
+
+    @classmethod
+    def download_audiofile(cls, metaId):
+        url = cls._get_request(path_param=cls.by_metaId + metaId)
+        return url
+
+    @classmethod
+    def search_audiofiles(cls):
```

Comment:
pls merge this to `search_audiofiles_by_tag`

Reply:
done